pkg/resource/template/fileStat_posix.go

```diff
@@ -9,22 +9,33 @@ import (
 	"io"
 	"os"
 	"syscall"
+
+	log "github.com/sirupsen/logrus"
 )
 
 // fileStat return a fileInfo describing the named file.
-func fileStat(name string) (fi fileInfo, err error) {
+func fileStat(name string) (fi fileInfo, ferr error) {
 	if isFileExist(name) {
 		f, err := os.Open(name)
 		if err != nil {
 			return fi, err
 		}
-		defer f.Close()
-		stats, _ := f.Stat()
+		defer func() {
+			if e := f.Close(); e != nil {
+				log.Error(e.Error())
```

Comment:
Generally this would be more like:  ` log.WithError(e).WithField("filename", name).Error("Error closing file")`